internal/database/offline/sqlutil/join.go

```diff
@@ -50,14 +50,20 @@ func DoJoin(ctx context.Context, dbOpt dbutil.DBOpt, opt DoJoinOpt) (*types.Join
 	emptyResult := &types.JoinResult{
 		Data: data,
 	}
-	// Step 1: prepare temporary table entity_rows
-	features := types.FeatureList{}
-	for _, featureList := range opt.FeatureMap {
-		features = append(features, featureList...)
+
+	featureNotEmpty := func() bool {
+		for _, features := range opt.FeatureMap {
+			if len(features) > 0 {
+				return true
+			}
+		}
+		return false
 	}
-	if len(features) == 0 {
+	if !featureNotEmpty() {
 		return emptyResult, nil
 	}
```

Comment:
This code is just to detect if there have features in `opt.FeatureMap`, but the `features` variable caused some confusion in reading the code (one would think it would be used somewhere down the line), so it was refactored with a temporary function.

Reply:
Maybe temporary function can be avoid?
```go
	empty := true
	for _, features := range opt.FeatureMap {
		if len(features) > 0 {
			empty = false
			break
		}
	}
	if empty {
		return &types.JoinResult{ Data: data }, nil
	}
```